src/services/resource-creator.js

```diff
@@ -5,55 +5,85 @@ const { ErrorHTTP422 } = require('./errors');
 const ResourceGetter = require('./resource-getter');
 const CompositeKeysManager = require('./composite-keys-manager');
 
-function ResourceCreator(model, params) {
-  const schema = Interface.Schemas.schemas[model.name];
-
-  this.perform = function perform() {
-    const promises = [];
-    const recordCreated = model.build(params);
-
-    if (model.associations) {
-      _.forOwn(model.associations, (association, name) => {
-        if (association.associationType === 'BelongsTo') {
-          promises.push(recordCreated[`set${_.upperFirst(name)}`](params[name], { save: false }));
-        }
-      });
+const getSetterName = (associationType, name) => {
+  let setterName;
+  if (associationType === 'HasOne') {
+    setterName = `set${_.upperFirst(name)}`;
+  } else if (['BelongsToMany', 'HasMany'].includes(associationType)) {
+    setterName = `add${_.upperFirst(name)}`;
+  }
+  return setterName;
+};
+
+class ResourceCreator {
+  constructor(model, params) {
+    this.model = model;
+    this.params = params;
+    this.schema = Interface.Schemas.schemas[model.name];
+  }
+
+  getPromisesBeforeSave(record) {
+    return (promises, [name, association]) => {
+      if (association.associationType === 'BelongsTo') {
+        const setterName = `set${_.upperFirst(name)}`;
+        const promise = record[setterName](this.params[name], { save: false });
+        promises.push(promise);
+      }
+      return promises;
+    };
+  }
+
+  getPromisesAfterSave(record) {
+    return (promises, [name, association]) => {
+      const setterName = getSetterName(association.associationType, name);
+      if (setterName) {
+        const promise = record[setterName](this.params[name]);
+        promises.push(promise);
+      }
+      return promises;
+    };
+  }
+
+  async handleSave(record, callback) {
+    const { associations } = this.model;
+    if (associations) {
+      callback = callback.bind(this);
+      const promisesBeforeSave = Object.entries(associations).reduce(callback(record), []);
+      await P.all(promisesBeforeSave);
+    }
+  }
+
+  async perform() {
+    // buildInstance
```

Comment:
I was thinking of really creating the methods named buildInstance, handleAssociationsBeforeSave, saveInstance, etc.

Reply:
There is only 16 lines of code, so I think it should be ok :)